grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -197,6 +258,13 @@ def _AddPrimitiveTypesSchemas(self) -> None:
       cast(Dict[str, Dict[str, str]], primitive_types_schemas)
     )
 
+  def _AddRDFTypesSchemas(self) -> None:
+    """Adds the OpenAPI schemas for RDF types."""
+    if self.schema_objs is None:
+      raise AssertionError("OpenAPI type schemas not initialized.")
+
+    self.schema_objs.update(rdf_types_schemas)
```

Comment:
This has more assertion logic than actual content. I think it is a right thing to do in this PR for consistency with the other initialization method, but I would keep in mind for the future to refactor these two (`schema_objs` is pretty static IIUC, why not make it a top-level constant)?

Reply:
Thank you for pointing this out! I will refactor them in a future PR. 

Regarding `schema_objs`, I kept it tied to the handler instance because it depends on the api router which is passed as an argument to the handler's constructor and from which the router method's args types and result types are traversed during schemas creation. I believe, but I'm not completely sure, that multiple routers can be used "at the same time" for requests from different users from what I've read in `http_api.py` [[1]](https://github.com/google/grr/blob/b472626b6ff4d05b24269a7ce52419d00a847607/grr/server/grr_response_server/gui/http_api.py#L423) [[2]](https://github.com/google/grr/blob/b472626b6ff4d05b24269a7ce52419d00a847607/grr/server/grr_response_server/gui/http_api.py#L191). If we had a single, top-level `schema_objs`, the OpenAPI description might expose too many/few API methods depending on the user maybe? Please let me know if I should move it to top-level.